File: src/main/java/com/github/joelws/release/tracker/interfaces/BusinessService.java
package com.github.joelws.release.tracker.interfaces;

import org.springframework.transaction.annotation.Transactional;

import javax.ws.rs.core.Response;

public interface BusinessService<I> {
    @Transactional
    Response create(final String json);

    @Transactional
    Response read(final I identifier);

    @Transactional
    Response update(final String json);

    @Transactional
    Response delete(final I identifier);

    @Transactional
    Response list();
}

File: src/main/kotlin/com/github/joelws/release/tracker/interfaces/BusinessService.kt
package com.github.joelws.release.tracker.interfaces

import org.springframework.transaction.annotation.Transactional

import javax.ws.rs.core.Response

interface BusinessService<in I> {

    @Transactional
    fun create(json: String): Response

    @Transactional
    fun read(identifier: I): Response

    @Transactional
    fun update(json: String): Response

    @Transactional
    fun delete(identifier: I): Response

    @Transactional
    fun list(): Response
}
